test/quantization/quantize_/workflows/int4/test_int4_tensor.py

```diff
@@ -5,22 +5,63 @@
 # LICENSE file in the root directory of this source tree.
 
 import unittest
+from dataclasses import dataclass
 
 import torch
 from torch.testing._internal.common_utils import (
     TestCase,
+    instantiate_parametrized_tests,
+    parametrize,
     run_tests,
 )
 
-from torchao.quantization import (
-    Int4WeightOnlyConfig,
-    quantize_,
+from torchao.core.config import AOBaseConfig
+from torchao.quantization import Int4WeightOnlyConfig, quantize_
+from torchao.quantization.transform_module import (
+    _QUANTIZE_CONFIG_HANDLER,
+    register_quantize_module_handler,
 )
 from torchao.quantization.utils import compute_error
-from torchao.utils import (
-    TORCH_VERSION_AT_LEAST_2_8,
-    is_sm_at_least_90,
-)
+from torchao.testing.model_architectures import Experts
+from torchao.utils import TORCH_VERSION_AT_LEAST_2_8, DummyModule, is_sm_at_least_90
+
+
+@dataclass
+class TestOnlyMoEQuantConfig(AOBaseConfig):
+    """Configuration for applying quantization to MoE
+    Args:
+        `base_config`: normal AO Config
+    """
+
+    base_config: AOBaseConfig
+
+
+def _moe_quant_tensor(weight, config):
+    def _moe_quant_tensor_base(weight, config):
+        base_config_handler = _QUANTIZE_CONFIG_HANDLER[type(config.base_config)]
+        dummy_mod = DummyModule(weight)
+        quant_mod = base_config_handler(dummy_mod, config.base_config)
+        return quant_mod.weight
+
+    return _moe_quant_tensor_base(weight, config)
+
+
+@register_quantize_module_handler(TestOnlyMoEQuantConfig)
+def moe_quant_fn(module, config: TestOnlyMoEQuantConfig):
```

Comment:
this is really confusing, could you share the result of `print(model)` after this function has been applied?

if it's going to print model with parameters wrapped in `Int4Tensor`, can we just wrap the parameters directly without all of these layers of abstraction?

Reply:
if this is working around the fact that `quantize_` needs to work on modules, IMO we should change quantize_ to handle this instead of working around? seems important for MoEs.

Reply:
yeah the parameters are wrapped in Int4Tensor, this is just applying quantization to each of the moe weights: w1, w2 and w3

I can inline these for now. can follow up with how to have an API for weights + configs separately